src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBulkCopyColumnOrderHintCollection.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Data.SqlClient
+{
+    /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/SqlBulkCopyColumnOrderHintCollection/*'/>
+    public sealed class SqlBulkCopyColumnOrderHintCollection : CollectionBase
+    {
+        private readonly HashSet<string> _columnNames = new HashSet<string>();
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/Item/*'/>
+        public SqlBulkCopyColumnOrderHint this[int index] => (SqlBulkCopyColumnOrderHint)List[index];
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/Add[@name="columnOrderHintParameter"]/*'/>
+        public SqlBulkCopyColumnOrderHint Add(SqlBulkCopyColumnOrderHint columnOrderHint)
+        {
+            if (columnOrderHint == null)
+            {
+                throw new ArgumentNullException(nameof(columnOrderHint));
+            }
+            if (string.IsNullOrEmpty(columnOrderHint.Column) ||
+                columnOrderHint.SortOrder == SortOrder.Unspecified)
+            {
+                throw SQL.BulkLoadInvalidOrderHint();
+            }
+            RegisterColumnName(columnOrderHint, columnOrderHint.Column);
+            InnerList.Add(columnOrderHint);
+            return columnOrderHint;
+        }
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/Add[@name="columnStringAndsortOrderSortOrder"]/*'/>
+        public SqlBulkCopyColumnOrderHint Add(string column, SortOrder sortOrder) => Add(new SqlBulkCopyColumnOrderHint(column, sortOrder));
+
+#pragma warning disable CS1591 // XML comment not needed. Method is not part of public API and class is sealed
+        protected override void OnClear()
+#pragma warning restore CS1591 
```

Comment:
I think we can add inline documentation instead of disabling the warning as below upside the function:
```
/// <summary>
/// 
/// </summary>
```
What do you think @cheenamalhotra?

Reply:
Doesn't hurt.. disabling warning should be last savior.